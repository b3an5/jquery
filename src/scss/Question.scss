.question-card {
  border: 3px solid lightblue;
  border-radius: 15px;
  background-color: lightgreen;
  padding: 10px;
  margin: 10px;
  position: relative;
  text-align: center;
}

.question-title {
  font-size: 3em;
  text-align: center;
  margin-top: 0px;
}

.vanilla-js {
  text-align: center;
}

.incorrect-answer {
  text-align: center;
  color: red;
}

.correct-answer {
  text-align: center;
  color: green
}

.checkmark {
  font-size: 5em;
  color: green;
  position: absolute;
  top: 0;
  left: 0;
}

.xmark {
  font-size: 5em;
  color:red;
  position: absolute;
  top: 15%;
  left: 0;
}

.answer-input {
  height: 50px;
  width: 500px;
  font-size: 2.5em;
  text-align: center;
  margin-top: 25px;
}

.question-button {
  display: block;
  text-align: center;
  margin: 10px auto;
  width: 80px;
  height: 40px;
  border: 1px solid black;
  border-radius: 10px;
  background-color: yellow;
}
