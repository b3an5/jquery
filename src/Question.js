import React, { Component } from "react";
import './scss/main.scss'

export default class Question extends Component {
  constructor() {
    super();
    this.state = {
      answerInput: '',
      answerVal: ''
    }
  }

  recordAnswer = (event) => {
    this.setState({
      answerInput: event.target.value
    })
  }

  enterAnswer = () => {
    if (this.props.data.jquery.includes(this.state.answerInput)) {
      this.setState({
        answerVal: 'correct'
      })
      this.props.scoreUp();
    } else {
      this.setState({
        answerVal: 'incorrect'
      })
    }
  }

  render() {
    let split = this.props.data.vanilla.split('\n');
    if(this.state.answerVal === '') {
      return(
        <div className='question-card'>
          <h1 className='question-title'>{this.props.data.title}</h1>
          {
            split.map((line) => {
              return <h2 className='vanilla-js'>{line}</h2>
            })
          }
          <input onChange={this.recordAnswer} typeof={Text} className='answer-input'></input>
          <button className='question-button' onClick={this.enterAnswer} >enter</button>
        </div>
      )
    } else if (this.state.answerVal === 'correct') {
      return (
        <div className='question-card'>
          <h1 className='question-title'>{this.props.data.title}</h1>
          <h2 className='vanilla-js'>{this.props.data.vanilla}</h2>
          <h1 className='correct-answer' >{this.props.data.jquery}</h1>
          <h1 className='checkmark'>&#10004;</h1>
        </div>
      )
    } else if (this.state.answerVal === 'incorrect') {
      return (
        <div className='question-card'>
          <h1 className='question-title'>{this.props.data.title}</h1>
          <h2 className='vanilla-js'>{this.props.data.vanilla}</h2>
          <h1 className='incorrect-answer' >{this.state.answerInput}</h1>
          <h1 className='correct-answer' >{this.props.data.jquery}</h1>
          <h1 className='xmark'>X</h1>
        </div>
      )
    }
  }
}