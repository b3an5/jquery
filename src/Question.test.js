import React from 'react';
import Question from './Question.js';
import { shallow } from 'enzyme';

const mockQuestionData = {
  "title": "Add Class",
  "jquery": "$(el).addClass(className);",
  "vanilla": `if (el.classList)
      el.classList.add(className);\nelse\n  el.className += ' ' + className;`,
  "id": 3
}

describe('Question', () => {

  let wrapper 
  const mockScoreUp = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <Question
      data={mockQuestionData}
      scoreUp={mockScoreUp} />
    )
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should have the correct default state', () => {
    expect(wrapper.state()).toEqual({
      answerInput: '',
      answerVal: ''
    })
  })

  it('should change answerInput in state when input is typed in', () => {
    expect(wrapper.state()).toEqual({ answerInput: '', answerVal: '' })
    wrapper.find('.answer-input').simulate('change', { target: { value: 'blah' } })
    expect(wrapper.state()).toEqual({ answerInput: 'blah', answerVal: '' })
  })

  it('should change answerVal in state when button is pressed and should read correct', () => {
    expect(wrapper.state()).toEqual({ answerInput: '', answerVal: '' })
    wrapper.find('.answer-input').simulate('change', { target: { value: 'blah' } })
    expect(wrapper.state()).toEqual({ answerInput: 'blah', answerVal: '' })
    wrapper.find('.question-button').simulate('click')
    expect(wrapper.state()).toEqual({ answerInput: 'blah', answerVal: 'incorrect' })
  })
  
  it('should change answerVal in state when button is pressed and should read correct', () => {
    expect(wrapper.state()).toEqual({ answerInput: '', answerVal: '' })
    wrapper.find('.answer-input').simulate('change', { target: { value: '$(el).addClass(className);' } })
    expect(wrapper.state()).toEqual({ answerInput: '$(el).addClass(className);', answerVal: '' })
    wrapper.find('.question-button').simulate('click')
    expect(wrapper.state()).toEqual({ answerInput: '$(el).addClass(className);', answerVal: 'correct' })
  })
})