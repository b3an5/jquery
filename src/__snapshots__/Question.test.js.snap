// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Question
    data={
      Object {
        "id": 3,
        "jquery": "$(el).addClass(className);",
        "title": "Add Class",
        "vanilla": "if (el.classList)
      el.classList.add(className);
else
  el.className += ' ' + className;",
      }
    }
    scoreUp={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="question-title"
        >
          Add Class
        </h1>,
        Array [
          <h2
            className="vanilla-js"
          >
            if (el.classList)
          </h2>,
          <h2
            className="vanilla-js"
          >
                  el.classList.add(className);
          </h2>,
          <h2
            className="vanilla-js"
          >
            else
          </h2>,
          <h2
            className="vanilla-js"
          >
              el.className += ' ' + className;
          </h2>,
        ],
        <input
          className="answer-input"
          onChange={[Function]}
          typeof={[Function]}
        />,
        <button
          className="question-button"
          onClick={[Function]}
        >
          enter
        </button>,
      ],
      "className": "question-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add Class",
          "className": "question-title",
        },
        "ref": null,
        "rendered": "Add Class",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "if (el.classList)",
          "className": "vanilla-js",
        },
        "ref": null,
        "rendered": "if (el.classList)",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "      el.classList.add(className);",
          "className": "vanilla-js",
        },
        "ref": null,
        "rendered": "      el.classList.add(className);",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "else",
          "className": "vanilla-js",
        },
        "ref": null,
        "rendered": "else",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "  el.className += ' ' + className;",
          "className": "vanilla-js",
        },
        "ref": null,
        "rendered": "  el.className += ' ' + className;",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "answer-input",
          "onChange": [Function],
          "typeof": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "enter",
          "className": "question-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "enter",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="question-title"
          >
            Add Class
          </h1>,
          Array [
            <h2
              className="vanilla-js"
            >
              if (el.classList)
            </h2>,
            <h2
              className="vanilla-js"
            >
                    el.classList.add(className);
            </h2>,
            <h2
              className="vanilla-js"
            >
              else
            </h2>,
            <h2
              className="vanilla-js"
            >
                el.className += ' ' + className;
            </h2>,
          ],
          <input
            className="answer-input"
            onChange={[Function]}
            typeof={[Function]}
          />,
          <button
            className="question-button"
            onClick={[Function]}
          >
            enter
          </button>,
        ],
        "className": "question-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add Class",
            "className": "question-title",
          },
          "ref": null,
          "rendered": "Add Class",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "if (el.classList)",
            "className": "vanilla-js",
          },
          "ref": null,
          "rendered": "if (el.classList)",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "      el.classList.add(className);",
            "className": "vanilla-js",
          },
          "ref": null,
          "rendered": "      el.classList.add(className);",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "else",
            "className": "vanilla-js",
          },
          "ref": null,
          "rendered": "else",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "  el.className += ' ' + className;",
            "className": "vanilla-js",
          },
          "ref": null,
          "rendered": "  el.className += ' ' + className;",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "answer-input",
            "onChange": [Function],
            "typeof": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "enter",
            "className": "question-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "enter",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
